firefox-ios/firefox-ios-tests/Tests/XCUITests/CreditCardsTests.swift

```diff
@@ -134,27 +134,19 @@ class CreditCardsTests: BaseTestCase {
         }
         XCTAssertEqual(saveAndFillPaymentMethodsSwitch.value! as! String, "0")
         app.buttons[creditCardsStaticTexts.AutoFillCreditCard.addCard].tap()
-        addCreditCard(name: "Test", cardNumber: cards[0], expirationDate: "0540")
+        let dateFiveYearsFromNow = Calendar.current.date(byAdding: .year, value: 5, to: Date())
+        let formatter = DateFormatter()
+        formatter.dateFormat = "MMyy"
+        let futureExpiryMonthYear = formatter.string(from: dateFiveYearsFromNow!)
+        addCreditCard(name: "Test", cardNumber: cards[0], expirationDate: futureExpiryMonthYear)
         if #available(iOS 16, *) {
             navigator.goto(NewTabScreen) // Not working on iOS 15
-            navigator.openURL("https://checkout.stripe.dev/preview")
+            navigator.openURL("https://mozilla.github.io/form-fill-examples/basic_cc.html")
             waitUntilPageLoad()
             // The autofill option (Use saved card prompt) is not displayed
-            let cardNumber = app.webViews["Web content"].textFields["Card number"]
-            app.swipeUp()
-            app.swipeUp()
-            mozWaitForElementToExist(cardNumber)
-            if !cardNumber.isHittable {
-                swipeUp(nrOfSwipes: 2)
-            }
-            cardNumber.tapOnApp()
-            let menuButton = AccessibilityIdentifiers.Toolbar.settingsMenuButton
-            if !app.buttons[menuButton].isHittable {
-                cardNumber.tapOnApp()
-            }
+            let cardNumber = app.webViews["Web content"].textFields["Card Number:"]
+            cardNumber.waitAndTap()
```

Comment:
I think after this we need to dismiss the keyboard.. the test is failing in Bitrise. It shows this screen:
<img width="142" alt="Screenshot 2024-10-25 at 10 15 15" src="https://github.com/user-attachments/assets/540a9d2b-0fca-4ba6-9ac5-b710348442e3">
and fails trying to go to the 'TabToolbar.menuButton'

Reply:
I run the test locally and it also fails. By adding a new line after that:
`cardNumber.tap()`
it works, so somehow the `waitAndTap()` is not working completely well here.... :(

Reply:
have you tried tapping Done button? doesn't that dismiss the keyboard?

Reply:
Even though you dismiss the keyboard, there will be another issue because the problem is that the option with the saved card is not offered since it does not detect the click in the Number field

Reply:
@isabelrios @dragosb01 
Test should now pass with or without open keyboard
Added in the commit here. Test passing for me locally now.
[e2e9ee4059fc2222ce2b2e4765f81099b49d5528](https://github.com/mozilla-mobile/firefox-ios/pull/22758/commits/e2e9ee4059fc2222ce2b2e4765f81099b49d5528)